elastic/eui
# [EuiBreadcrumb] Fix styling on `application` types with popovers + perf improvements

## Summary

closes https://github.com/elastic/eui/issues/7577

| Before | After |
|--------|--------|
| <img width="323" alt="" src="https://github.com/elastic/eui/assets/549407/453c9855-67bd-4e9d-a7da-ee5d7c0f43e6"> | <img width="328" alt="" src="https://github.com/elastic/eui/assets/549407/e39eeb75-197f-4a2f-a862-a23334cf0239"> |

The underlying fix is to separate out the breadcrumb CSS from the truncation CSS and passing them to separate DOM elements for breadcrumbs with popovers. The final diff is a bit hard to follow because I ended up just pulling out breadcrumbs with popovers to a separate subcomponent, apologies for that 😬 

## QA

- Go to https://eui.elastic.co/pr_7580/storybook/?path=/story/navigation-euibreadcrumbs--playground&args=type:application
- Change the `max` control to `2`
- [x] Confirm the `...` collapsed button does not look broken
- Expand the `breadcrumbs` array and toggle the last breadcrumb. Click the `+` button and add a field: `popoverContent: 'anything'`
- [x] Confirm the popover toggle does not look broken

### General checklist

- Browser QA
- Docs site QA - N/A
- Code quality checklist - N/A, primarily a visual/CSS bug - snapshots should be mostly the same as before
- Release checklist
- Designer checklist - N/A
